voltixApp/VoltixApp/Views/Keygen/KeyGenSummaryView.swift

```diff
@@ -0,0 +1,180 @@
+//
+//  KeyGenSummaryView.swift
+//  VoltixApp
+//
+//  Created by Amol Kumar on 2024-04-11.
+//
+
+import SwiftUI
+
+struct KeyGenSummaryView: View {
+    @ObservedObject var viewModel: KeygenPeerDiscoveryViewModel
+    
+    @State var numberOfMainDevices = 0
+    @State var numberOfBackupDevices = 0
+    
+    var body: some View {
+        ZStack {
+            Background()
+            view
+        }
+        .navigationTitle(NSLocalizedString("summary", comment: "Summary"))
+        .onAppear {
+            setData()
+        }
+    }
+    
+    var view: some View {
+        VStack {
+            content
+            button
+        }
+    }
+    
+    var content: some View {
+        ScrollView {
+            VStack(spacing: 32) {
+                header
+                devicesList
+                disclaimers
+            }
+            .padding(.top, 30)
+            .padding(.horizontal, 22)
+            .foregroundColor(.neutral0)
+        }
+    }
+    
+    var header: some View {
+        VStack(spacing: 12) {
+            title
+            numberOfDevicesTitle
+        }
+    }
+    
+    var devicesList: some View {
+        VStack(spacing: 16) {
+            devicesListTitle
+            list
+        }
+    }
+    
+    var title: some View {
+        Text(NSLocalizedString("youAreCreatingA", comment: "You are creating a"))
+            .font(.body12MontserratSemiBold)
+    }
+    
+    var numberOfDevicesTitle: some View {
+        Group {
+            Text("\(numberOfMainDevices) ") +
+            Text(NSLocalizedString("of", comment: "of")) +
+            Text(" \(numberOfMainDevices+numberOfBackupDevices) ") +
+            Text(NSLocalizedString("vaults", comment: "vaults"))
+        }
+        .font(.body20MontserratSemiBold)
+    }
+    
+    var devicesListTitle: some View {
+        Text(NSLocalizedString("withTheseDevices", comment: "With these devices"))
+            .font(.body12MontserratSemiBold)
+    }
+    
+    var list: some View {
+        var index = 0
+        var pairDevices = numberOfMainDevices
+        
+        return VStack(spacing: 16) {
+            ForEach(Array(viewModel.selections), id: \.self) { selection in
+                index += 1
+                pairDevices -= selection==Utils.getLocalDeviceIdentity() ? 0 : 1
```

Comment:
can you change `Utils.getLocalDeviceIdentity()` -> `viewModel.localPartyID` instead? 
If it is reshare, the localPartyID might be different